src/api/combinedGrantsExt.ts

```diff
@@ -0,0 +1,62 @@
+import {
+    defaultTableFilter,
+    SortingProps,
+    supabaseClient,
+    TABLES,
+} from 'services/supabase';
+import { Grants } from 'types';
+
+// Used when getting grants for a user to know what tenant to list
+const getGrantsForUser = (userId: string, adminOnly?: boolean) => {
+    let queryBuilder = supabaseClient
+        .from<Grants>(TABLES.COMBINED_GRANTS_EXT)
+        .select(`*`);
+
+    if (adminOnly) {
+        queryBuilder = queryBuilder.eq('capability', 'admin');
+    }
+
+    return queryBuilder.eq('user_id', userId);
+};
+
+// Used to display all the grants for everything in the admin page
+const getGrantsForEverything = (
```

Comment:
The query helper naming convention will take a few iterations to see what _feels right_, so I am comfortable moving forward with the original function name. That said, the naming convention applied to the hydration-related query helpers is as follows:

**Differentiating queries that pull the same classification of data from a table (e.g., endpoint schema and/or resource schema from the `connector_tags` table):**

> `getDataClassification` -- an overarching query that pulls multiple forms of a data classification (e.g., both the resource and endpoint schemas from the `connector_tags` table).
> 
> `getDataClassification_SpecificIdentifier` -- a query that pulls a single form of a data classification (e.g., either the resource schema or the endpoint schema from the `connector_tags` table).

This was inspired by the store selector hook naming convention. It can be observed by the `getSchema_Endpoint` and `getSchema_Resource` helpers in the aforementioned file.

<br />

**Differentiating queries that pull data from the same column(s) of a table (e.g., the `spec` column of the `live_specs_ext` table):**

>`getDataClassificationByFocalColumn` -- a query that pulls data from a specific column(s) of a table where the _focal column_ is the distinguishing filter applied.

With those definitions in mind, `getGrants` is a suitable name alternative for `getGrantsForEverything` since the more nuanced queries identify the _use-case_ or distinguishing factor in the name. I would _recommend_ that adjustment. As far as the other query helper names go, I feel the `getDataClassification_SpecificIdentifier` definition is most applicable since you are targeting the `Grant` type explicitly (i.e., `getGrants_User` and `getGrants_AuthToken`). _Thinking out loud_.

Reply:
I actually only called is that because it was the last query I found while working through these and was "Oh crap... this should be called getGrants".

Reply:
I am still not sold on using the selector hooks naming pattern with apis. I played with that name with these and did not like it as much. But agree - it'll take us awhile to figure out how we're gonna name and structure stuff with the different approach.